d3a_api_client/redis_client_base.py

```diff
@@ -2,57 +2,73 @@
-    def wrapped(self, *args, **kwargs):
-        if not self.is_active:
-            raise RedisAPIException(f'Registration has not completed yet.')
-        return f(self, *args, **kwargs)
-
-    return wrapped
-
-
-class RedisClient(APIClientInterface):
+class RedisClientBase(APIClientInterface):
     def __init__(self, area_id, autoregister=True, redis_url='redis://localhost:6379',
                  pubsub_thread=None):
         super().__init__(area_id, autoregister, redis_url)
+        self.area_uuid = None
+        self.area_slug = slugify(area_id, to_lower=True)
         self.redis_db = StrictRedis.from_url(redis_url)
         self.pubsub = self.redis_db.pubsub() if pubsub_thread is None else pubsub_thread
         self.area_id = area_id
         self.device_uuid = None
         self.is_active = False
         self._blocking_command_responses = {}
+        self._transaction_id_buffer = []
+        self._subscribed_aggregator_response_cb = None
         self._subscribe_to_response_channels(pubsub_thread)
         self.executor = ThreadPoolExecutor(max_workers=MAX_WORKER_THREADS)
         if autoregister:
             self.register(is_blocking=True)
 
+    @property
+    def _channel_prefix(self):
+        return f"{self.area_slug}"
+
     @property
     def channel_subs(self):
         return {
-            f"{self.area_id}/response/register_participant": self._on_register,
-            f"{self.area_id}/response/unregister_participant": self._on_unregister,
+            f'{self._channel_prefix}/response/register_participant': self._on_register,
+            f'{self._channel_prefix}/response/unregister_participant': self._on_unregister,
+            f'{self._channel_prefix}/*': self._on_event_or_response
         }
 
     def _subscribe_to_response_channels(self, pubsub_thread=None):
         self.pubsub.psubscribe(**self.channel_subs)
+        if b'aggregator_response' in self.pubsub.patterns:
+            self._subscribed_aggregator_response_cb = self.pubsub.patterns[b'aggregator_response']
         if pubsub_thread is None:
             self.pubsub.run_in_thread(daemon=True)
 
-    def register(self, is_blocking=False):
+    def _check_buffer_message_matching_command_and_id(self, message):
+        if key_in_dict_and_not_none(message, "transaction_id"):
+            transaction_id = message["transaction_id"]
+            if not any(command in ["register", "unregister"] and "transaction_id" in data and
+                       data["transaction_id"] == transaction_id
+                       for command, data in self._blocking_command_responses.items()):
+                raise RedisAPIException("There is no matching command response in _blocking_command_responses.")
+        else:
+            raise RedisAPIException(
+                "The answer message does not contain a valid 'transaction_id' member.")
+
+    def _check_transaction_id_cached_out(self, transaction_id):
+        return transaction_id in self._transaction_id_buffer
```

Comment:
Transactions are never popped out of `self._transaction_id_buffer` in the new `RedisClientBase`, right?

Reply:
Indeed! I fix this in the frame of my comment above https://github.com/gridsingularity/d3a-api-client/pull/134#discussion_r618159818